File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.py
class Solution(object):
    def numberOfSubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        
        lasta = -1
        lastb = -1
        lastc = -1
        result = 0 

        for i in range( 0 , len(s) ):
            if( s[i] == 'a' ):
                lasta = i 
            elif( s[i] == 'b' ):
                lastb = i 
            else:
                lastc = i 

            result += min( lasta , lastb , lastc ) + 1 


        return result             

File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var numberOfSubstrings = function(s) {
    
    let lasta = -1 , lastb = -1 , lastc = -1 , result = 0 ;

    for( let i = 0 ; i < s.length ; i++ ){

            if( s[i] == 'a' ) lasta = i ;
            else if( s[i] == 'b' ) lastb = i ;
            else{
                lastc = i ;
            }

            if( lasta != -1 && lastb != -1 && lastc != -1 ){
                result += Number(Math.min(  lasta , lastb , lastc ) + 1) ;
            }
        }

    return Number(result) ;
};  